src/core/ext/transport/cronet/transport/cronet_transport.cc

```diff
@@ -1096,16 +1100,24 @@ static enum e_op_result execute_stream_op(struct op_and_state* oas) {
         /* Should never reach here */
         GPR_ASSERT(false);
       }
-      if (GRPC_ERROR_NONE !=
+      while (write_slice_buffer.length <
+             stream_op->payload->send_message.send_message->length()) {
+        /*
+          TODO(roth): When we add support for incremental sending,
+          this code will need to be changed to support asynchronous delivery
+          of the send_message payload.
+        */
+        if (GRPC_ERROR_NONE !=
```

Comment:
The call to `Next()` should be moved inside of this loop.  There should be one call to `Next()` before each call to `Pull()`.

Reply:
done